basis/tools/scaffold/scaffold.factor

```diff
@@ -272,7 +273,19 @@ PRIVATE>
 : scaffold-platforms ( vocab platforms -- )
     [ "platforms.txt" ] dip scaffold-metadata ;
 
+: vocab-root-index ( vocab-root -- i/f )
+    vocab-roots get index ;
```

Comment:
I probably wouldn't make this a new word since it's only used in one place...

Reply:
Also, isn't this a simpler shadow check to just see if the vocab name has a ``find-vocab-root`` already?

```factor
ERROR: shadowed-vocab name root ;

: check-shadowed ( string -- )
    dup find-vocab-root [ shadowed-vocab ] [ drop ] if* ;
```

Reply:
Or maybe, since some vocabularies like ``accessors`` and ``cocoa.classes`` are dynamic, this is better just to check if it exists:

```factor
ERROR: vocab-exists name ;

: check-vocab-exists ( name -- name )
    dup vocab-exists? [ vocab-exists ] when ;
```

Reply:
@mrjbq7: "I probably wouldn't make this a new word since it's only used in one place..."
It's called twice, so it's more readable this way, no?

Regarding the `find-vocab-root`, the idea is that we want to prevent scaffolding of vocabs that would already be shadowed on creation (see #100), so that reloading them would fail. If, on the other hand, we want to create a vocab that shadows an existing one - that's not a error, see unit-tests.

Reply:
I saw that in the unit tests, but I think maybe that we should be more conservative, to not allow creating a shadowed vocab or a shadowing vocab, no?

Reply:
@mrjbq7: maybe you're right, and the conservative approach is better.

I'm not sure the `vocab-exists` would work well for the error tuple. `vocab-exists?` clashes with the class checker word.

Reply:
Does it clash?  I'm using the ``vocabs.loader:vocab-exists?`` word.

Maybe it shouldn't be called ``check-vocab-exists`` since we have ``ensure-vocab-exists``, so maybe a different name like ``check-new-vocab`` or something.

Reply:
I'm just saying that we have explicitly defined `vocabs.loader:vocab-exists?` and implicitly (by creating a tuple) defined `tools.scaffold:vocab-exists?`. That creates ambiguity.
Or do you think it's OK to have that as long as the two vocabs are unlikely to be imported at the same time? I'm interested to know your approach to resolving such situations in general.

Reply:
Ooooooohhh, yes. 

Maybe better to say something easier for the user to understand like:

```factor
ERROR: vocab-must-not-exist name ;
```

Reply:
Thanks for the reply!

Are we waiting for more feedback here, or is it decided? Would you like to commit your proped code yourself, or would you like me to redo the PR?
I'm fine either way. I'm just glad we are reducing the issue count. : ))

Reply:
Sure, fixed in 68eb6833872571a8e931cf2246838e5e0e619052.